test/libcommonserver/io/testcheckdirectoryiterator.cpp

```diff
@@ -56,7 +56,7 @@ void TestIo::testCheckDirectoryIteratorNonExistingPath() {
 }
 
 void TestIo::testCheckDirectoryIteratorExistingPath() {
-    TemporaryDirectory tempDir;
```

Comment:
No, I wanted to create a similar class for remote side `RemoteTemporaryDirectory` but it was too much changes according to the scope of this PR.
However, I still think it would be good to implement such a utility method as you did for the local temporary directory.